src/psyclone/transformations.py

```diff
@@ -101,6 +112,16 @@ def _validate(self, node_list):
                     .format(str(child), child.position, prev_position))
             prev_position = child.position
 
+        # Check that the proposed region contains only supported node types
+        for child in node_list:
+            flat_list = [child] + child.walk(child.children, object)
```

Comment:
I called it that because I'm converting from a tree structure into a list, a process that is sometimes called 'flattening' (I believe) and also, `list` is a Python built-in.